mono/metadata/class-init.c

```diff
@@ -2904,105 +2886,114 @@ mono_class_setup_vtable_general (MonoClass *klass, MonoMethod **overrides, int o
 		mono_class_setup_vtable_full (klass->parent, in_setup);
 
 		if (mono_class_set_type_load_failure_causedby_class (klass, klass->parent, "Parent class failed to load"))
-			return;
+			return -1;
 
 		max_vtsize += klass->parent->vtable_size;
-		cur_slot = klass->parent->vtable_size;
+		*cur_slot = klass->parent->vtable_size;
 	}
 
 	max_vtsize += mono_class_get_method_count (klass);
 
 	/*Array have a slot for stelemref*/
 	if (mono_class_need_stelemref_method (klass)) {
-		stelemref_slot = cur_slot;
+		*stelemref_slot = *cur_slot;
 		++max_vtsize;
-		++cur_slot;
+		++*cur_slot;
 	}
+	return max_vtsize;
+}
 
-	/* printf ("METAINIT %s.%s\n", klass->name_space, klass->name); */
+static int
+mono_class_setup_vtable_ginst (MonoClass *klass, GList *in_setup, MonoError *error)
```

Comment:
1. return a `gboolean` instead of `int`
2. `/* LOCKING */` comment
3. comment like `/* returns FALSE and sets MonoError if setup fails */ `

but also this method is pretty weird because sometimes it returns "success" but really the `klass` is already marked with a failure.

I wonder if it should just have a local `MonoError` and do its own `fail` case and just always return (ie you won't need the `goto fail` code in the caller)